src/test/kotlin/no/skatteetaten/aurora/mokey/service/ManagementEndpointTest.kt

```diff
@@ -54,42 +58,38 @@ class ManagementEndpointTest : AbstractTest() {
         managementEndpoint.getInfoEndpointResponse()
     }
 
-    @TestFactory
-    fun `should handle management link response errors`(): Collection<DynamicTest> {
+    @ParameterizedTest(name = "{0} as {1} yields {2} for management endpoint")
+    @ArgumentsSource(TestDataProvider::class)
+    fun `should handle management link responses`(response: String, mediaType: MediaType, errorCode: String?, responseCode: HttpStatus) {
 
-        data class TestData(
-                val response: String,
-                val mediaType: MediaType,
-                val errorCode: String? = null,
-                val responseCode: HttpStatus = OK)
+        server.expect(once(), requestTo(managementUrl)).andRespond(withStatus(responseCode).body(response).contentType(mediaType))
 
-        return listOf(
-                TestData("{}", APPLICATION_JSON),
-                TestData("{ \"_links\": {}}", APPLICATION_JSON),
-                TestData("", APPLICATION_JSON),
-                TestData("{}", APPLICATION_JSON, responseCode = INTERNAL_SERVER_ERROR),
-                TestData("{ \"_links\": { \"health\": null}}", APPLICATION_JSON, "INVALID_FORMAT"),
-                TestData("{ _links: {}}", APPLICATION_JSON, "INVALID_JSON"),
-                TestData("", APPLICATION_JSON, "ERROR_404", NOT_FOUND),
-                TestData("<html><body><h1>hello</h1></body></html>", TEXT_HTML, "INVALID_JSON"),
-                TestData("<html><body><h1>hello</h1></body></html>", TEXT_HTML, "INVALID_JSON", INTERNAL_SERVER_ERROR),
-                TestData("<html><body><h1>hello</h1></body></html>", APPLICATION_JSON, "INVALID_JSON", INTERNAL_SERVER_ERROR)
-        ).map {
-            server.expect(once(), requestTo(managementUrl)).andRespond(withStatus(it.responseCode).body(it.response).contentType(it.mediaType))
+        val create: () -> Unit = { ManagementEndpoint.create(restTemplate, managementUrl) }
+        if (errorCode != "SUCCESS") {
```

Comment:
Jeg sliter litt med denne, flytlogikk i tester prøver jeg å unngå hvis mulig.
Syns også det er litt vanskelig å følge logikken her, for i en del tilfeller så vil ikke ManagementEndpoint.create bli kalt. Og i noen tilfeller, hvis det er SUCCESS, så kjøres ikke assertThat.

Et forslag fr ameg, jeg vet ikke om dette er aktuelt men bare en idé.
Kunne vi ha delt denne opp i 2 typer tester? En som tester "valid" responses og en som tester feil-situasjoner?

Reply:
Det kan vi veldig godt gjøre. Jeg tenkte tanken selv i går, men så "gadd" jeg ikke gjøre noe med det. Fikser det nå.